src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4712,6 +4708,49 @@ private void SetMethodGroupReceiverNullability(BoundExpression receiver, TypeWit
             _methodGroupReceiverMapOpt[receiver] = type;
         }
 
+        private MethodSymbol CheckMethodGroupReceiverNullability(BoundMethodGroup group, NamedTypeSymbol delegateType, MethodSymbol method, bool invokedAsExtensionMethod)
+        {
+            var receiverOpt = group.ReceiverOpt;
+            if (TryGetMethodGroupReceiverNullability(receiverOpt, out TypeWithState receiverType))
+            {
+                var syntax = group.Syntax;
+                if (!invokedAsExtensionMethod)
+                {
+                    method = (MethodSymbol)AsMemberOfType(receiverType.Type, method);
+                }
+                if (method.Arity > 0 && HasImplicitTypeArguments(group.Syntax))
+                {
+                    var binder = compilation.GetBinderFactory(syntax.SyntaxTree).GetBinder(syntax);
+                    var arguments = ArrayBuilder<BoundExpression>.GetInstance();
+                    if (invokedAsExtensionMethod)
+                    {
+                        arguments.Add(CreatePlaceholderIfNecessary(receiverOpt, receiverType.ToTypeWithAnnotations()));
+                    }
+                    // Create placeholders for the arguments. See Conversions.GetDelegateArguments()
+                    // which is used for that purpose in initial binding.
+                    foreach (var parameter in delegateType.DelegateInvokeMethod.Parameters)
+                    {
+                        var parameterType = parameter.TypeWithAnnotations;
+                        arguments.Add(new BoundExpressionWithNullability(syntax, new BoundParameter(syntax, parameter), parameterType.NullableAnnotation, parameterType.Type));
+                    }
+                    method = InferMethodTypeArguments(binder, method, arguments.ToImmutableAndFree(), argumentRefKindsOpt: default, argsToParamsOpt: default, expanded: false);
+                }
```

Comment:
>} [](start = 16, length = 1)

I wonder if annotation attributes on method parameters should be checked in some way (not blocking this PR)

Reply:
The annotations would only be applied when the delegate is invoked, not when the delegate is created.

---
In reply to: [282700235](https://github.com/dotnet/roslyn/pull/35601#discussion_r282700235) [](ancestors = 282700235)

Reply:
That seems strange. Let's say we do have a [MaybeNull] attribute. Can I convert `[MaybeNull] T M()` to `Func<T>` safely?

---
In reply to: [282755353](https://github.com/dotnet/roslyn/pull/35601#discussion_r282755353) [](ancestors = 282755353,282700235)

Reply:
The `[MaybeNull]` attribute is distinct from the other annotation attributes in that it directly represents the nullability of the type that it is applied to, in the same way that `?` does for a reference type. `[MaybeNull]` would be included in the `TypeWithAnnotations` so it would affect the delegate return type or parameter types.

---
In reply to: [282976456](https://github.com/dotnet/roslyn/pull/35601#discussion_r282976456) [](ancestors = 282976456,282755353,282700235)